lib/nerves.ex

```diff
@@ -1,6 +1,40 @@
 defmodule Nerves do
+  @elixir_version_req "~> 1.7"
+  @otp_version_req "~> 21.0"
+
   def version, do: unquote(Mix.Project.config()[:version])
   def elixir_version, do: unquote(System.version())
+  def otp_release, do: unquote(System.otp_release())
+
+  def system_requirements(elixir_version \\ nil, otp_release \\ nil) do
+    elixir_version = elixir_version || elixir_version()
+    otp_release = otp_release || otp_release()
+
+    with {:ok, otp_rel_version} <- Version.parse(otp_release <> ".0.0"),
+         true <- Version.match?(elixir_version, @elixir_version_req),
+         true <- Version.match?(otp_rel_version, @otp_version_req) do
+      :ok
+    else
+      _ ->
+        Nerves.Utils.Shell.warn("""
+        Nerves #{version()} requires
+        Elixir:      #{@elixir_version_req}
+        OTP release: #{@otp_version_req}
+
+        Your system has
+        Elixir:      #{elixir_version}
+        OTP release: #{otp_release}
+
+        Please resolve the issue by
+        * Installing a version of Elixir / OTP that is compatible with the
+          Minimal requirements.
+        * Pin your nerves and nerves_bootstrap dependencies to an older
+          version that supports your version of Elixir / OTP.
```

Comment:
```suggestion
        Nerves #{version()} requires at least Elixir #{@elixir_version_req} and Erlang/OTP #{@otp_version_req}.

        Your system has Elixir #{elixir_version} and Erlang/OTP #{otp_release}.

        Please resolve this by either:

        1. Installing a compatible version of Elixir and Erlang/OTP

        2. Pinning your nerves and nerves_bootstrap dependencies to
           older versions.
```

Here's what it looks like:

```
Nerves 1.5.2 requires at least Elixir 1.7 and Erlang/OTP 21.0.

Your system has Elixir 1.6.2 and Erlang/OTP 19.0.

Please resolve the issue by either:

1. Installing a compatible version of Elixir and Erlang/OTP

2. Pinning your nerves and nerves_bootstrap dependencies to
   older versions
```